Standardize cycle txs/joe

ahmxdiqbal opened the pull request.

ahmxdiqbal committed:
updated join cycle tx type

ahmxdiqbal committed:
refactor/change KeepInStandby to StandbyRefreshRequest

magonjr committed:
StandbyAddRequest

TieDyedSheep committed:
SYS-164 added workflow to run linter and trigger smoke tests (#1)

* SYS-164 added workflow to run linter and trigger smoke tests

* SYS-164 use node 18

TieDyedSheep committed:
SYS-136 remove jenkins trigger from types (#3)

afostr reviewed (commented):
these all look good but we can wait until the related larger set of changes is ready

TieDyedSheep committed:
SYS-183 update to call a reusable workflow (#4)

urnotsam committed:
add random data to cyclerecord type (#5)

asyed94 committed:
1.2.12

arhamj committed:
Adding safeStringify & safeJsonParser (#2)

Adding functionality for stringification and parsing

fix(util): added imports and simplified types. linting and formatting fixes

feat(ut): added base unit tests

feat(buffer): added tests for buffers

style(name): renamed parser function to parse for standardisation

feat(ut): added unit tests for parser and big fixes

feat(ts): removed test files from compilation

Fix typo in safeJsonParse

- Added missing optional reviver param
- Changed the return type to any
remove typeReviver from safeJsonParse
unit tests for nested bigInt arrays and error handling

fixed formatting and updated comments

feat(ut): added coverage tools

regenerated package-lock.json

feat(format): updated prettier config

ci(ut): added ut to github ci pipeline

feat(revert): revert to retain buffer behaviour

feat(u8a): added support for u8a

feat: export typeReviver so it can be used by core body parser

Co-authored-by: sabdulazeem01 <sabdulazeem01@gmail.com>
Co-authored-by: CombatPug <pe.ladar@icloud.com>

arhamj committed:
1.2.13 (#8)
